### PR TITLE
tsweb: add request ID for errors

### PR DESCRIPTION
**Pull request by sailorfrag**
If an optional request ID generating func is supplied to StdHandler, then requests that return an error will be logged with a request ID that is also shown as part of the response.

Updates tailscale/corp#2549

Change-Id: Ic7499706df42f95b6878d44d4aab253e2fc6a69b

### CODE REVIEW

File: tsweb/log.go
Comment (bradfitz) [inline tsweb/log.go:48]: I was going to say these docs aren't in the usual style (complete sentences starting with the name of the field) but then I noticed none of them are. At least fix this one and we can clean the others later?

File: tsweb/log.go
Comment (bradfitz) [inline tsweb/log.go:50]: Any guidance here or elsewhere on its format? I guess more than digits. Whitespace? What's the allowed alphabet of characters?

File: tsweb/tsweb.go
Comment (bradfitz) [inline tsweb/tsweb.go:173]: Maybe if we use a `type RequestID string` type we can put the docs about recommended format in one place

File: tsweb/tsweb.go
Comment (bradfitz) [inline tsweb/tsweb.go:314]: Yeah, the docs on the type might mention it's appended after "internal server error\n" so should have a prefix that sounds like a useful identifier for humans reporting bugs. (Having just a number or just hex would look weird.)

File: tsweb/log.go
Comment (sailorfrag) [reply tsweb/log.go]: Yeah it seemed weird but I went with matching the surrounding comment format.

I've fixed this one.

File: tsweb/log.go
Comment (sailorfrag) [reply tsweb/log.go]: I didn't want `tsweb` to be particularly opinionated. Anything printable is fine. But I took your suggestion regarding adding a type and documenting the format there.

File: tsweb/tsweb.go
Comment (sailorfrag) [reply tsweb/tsweb.go]: Done

File: tsweb/tsweb.go
Comment (sailorfrag) [reply tsweb/tsweb.go]: Included in the docs.

File: tsweb/log.go
Comment (bradfitz) [inline tsweb/log.go:51]: mention that it may be empty if not configured

**Review by bradfitz — APPROVED** [review]

**Comment by sailorfrag** [comment]
Alright, I'm going to hold off on merging this until the stable release is cut unless I hear a strong opinion that I should merge right away. I don't expect it to affect the release, but there's minimal cost in delaying for a few days.

**Pull request merged**
